arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/ScheduleTest.kt

```diff
@@ -214,39 +218,39 @@ class ScheduleTest : ArrowFxSpec(spec = {
   }
 })
 
-private fun fibs(one: Long): Sequence<Long> = generateSequence(Pair(0L, one)) { (a, b) ->
+private fun fibs(one: Double): Sequence<Double> = generateSequence(Pair(0.0, one)) { (a, b) ->
   Pair(b, (a + b))
 }.map { it.first }
 
-private fun exp(base: Long): Sequence<Long> = generateSequence(Pair(base, 1.0)) { (_, n) ->
-  Pair((base * 2.0.pow(n)).toLong(), n + 1)
+private fun exp(base: Double): Sequence<Double> = generateSequence(Pair(base, 1.0)) { (_, n) ->
+  Pair(base * 2.0.pow(n), n + 1)
 }.map { it.first }
 
-private fun linear(base: Long): Sequence<Long> = generateSequence(Pair(base, 1L)) { (_, n) ->
+private fun linear(base: Double): Sequence<Double> = generateSequence(Pair(base, 1.0)) { (_, n) ->
   Pair((base * n), (n + 1))
 }.map { it.first }
 
-private suspend fun <I, A> Schedule<I, A>.calculateSchedule1(input: I): Schedule.Decision<Any?, A> =
+private suspend fun <I : Number, A> Schedule<I, A>.calculateSchedule1(input: I): Schedule.Decision<I?, A> =
```

Comment:
I think the `ClassCastException` comes from these changes. I think these changes are unnecassery, I'd need to look deeper why this is `Any?` but I think it's necassary for these test.

Reply:
I believe it's any because we use an associated type in the ScheduleImpl class and thus it gets erased at runtime. The associated type makes the ScheduleImpl typesafe and much easier to use and since it rarely shows up for the user (only in the lowest level combinators). However the tests test the inner workings and as such need to provide type information, which our best bet is `Any?`. We can't assume the state (which is what this generic represents) to be in any way related to our input.

Reply:
Unfortunately, that doesn't fix it.
I've tried it before fixing it to `Any?`, those changes can be seen in this [commit](https://github.com/arrow-kt/arrow-fx/pull/359/commits/6e205ab37a4f0d8fe320ad64546dca5913003c75). 
To reproduce the error, I've singled out the other tests besides `Schedule.linear`

What I can observe is that the addition here:
https://github.com/arrow-kt/arrow-fx/blob/6e205ab37a4f0d8fe320ad64546dca5913003c75/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt#L652
never get's executed, so this might be an error from `kotlin.time.Duration`
In the past we would calculate addition like this:
https://github.com/arrow-kt/arrow-fx/blob/3ca8e1f27ab60af8ef8c8c3740686059eb5cbbc2/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Duration.kt#L23

Which passes our Tests, prior to this PR.
I am still investigating how to fix it